pyphare/pyphare/pharein/diagnostics.py

```diff
@@ -127,18 +126,30 @@ def __init__(self, name, **kwargs):
                 f"{self.__class__.__name__,}.flush_every cannot be negative"
             )
 
-        if any(
-            [
-                self.quantity == diagnostic.quantity
-                for diagnostic in global_vars.sim.diagnostics
-            ]
-        ):
-            raise RuntimeError(
-                f"Error: Diagnostic ({kwargs['quantity']}) already registered"
-            )
-
         self.__extent = None
-        global_vars.sim.add_diagnostics(self)
+
+        # if a diag already is registered we just contactenate the timestamps
+        addIt = True
+        registered_diags = global_vars.sim.diagnostics
+        for diagname, diag in registered_diags.items():
+            if self.quantity == diag.quantity:
+                print(
+                    f"{diag.name} already registered {self.quantity}, merging timestamps"
+                )
+                my_times = self.write_timestamps
```

Comment:
`my` -> `new`?

Reply:
I think the proposed change is misleading. 'my' refers to 'self', 'new' refers to future self.
adding self.write_timestamps in the np.concatenates makes the line quite long and I find

' new_times = np.concatenate((my_times, existing_times))` 

quite easy to read.

I don't think there is a perf issue as everything is just references anyway.

Reply:
`new` IMO refers to additional, if there is existing

I have an aversion to `my`, implies personhood